### PR TITLE
clean structure, add signature scanning

### PR DESCRIPTION
**Pull request by ero-qt**
sig scanning should remove all possibility of the script breaking with updates

### CODE REVIEW

**Comment by neesi** [comment]
Thanks for the script, very helpful. I wasn't even aware of signature scanning.

It looks like the framecounter pointer path (0x610, 0x354, 0x120) stops working every time the data.win file is updated unless you're lucky. In the last 10 versions, data.win has changed 6 times. Room pointer sig works through all 10.

I modified your script to obtain the framecounter address using a hook instead. The framecounter signature has changed only once in the last 10 versions (after a major update) and levelactive sig has not changed at all.

I won't merge this yet in case you want to take a look and use a better method / fix things.

File: LOVE/LOVE_livesplit.asl [hunk 1]
```diff
@@ -2,49 +2,214 @@ state("Love") {}
 
 startup
 {
+	vars.ScanThreadReady = false;
 	vars.Dbg = (Action<dynamic>) ((output) => print("[LOVE ASL] " + output));
 
-	settings.Add("IL_Splits_LOVE", true, "Enable automatic splits for IL mode.");
+	settings.Add("", false);
```
Comment (ero-qt) [inline LOVE/LOVE_livesplit.asl]: Stop doing this. The settings are not for this. If you need to tell the user some info, put it in a `README.md` on this repo, then link to that in the `LiveSplit.AutoSplitters.xml`.

File: LOVE/LOVE_livesplit.asl [hunk 2]
```diff
@@ -2,49 +2,214 @@ state("Love") {}
...
+	settings.Add("   Enter speedrun mode to activate the splitter.", false);
+	settings.Add("   After that, it will work in every other game mode.", false);
 }
 
 init
 {
+	vars.levelactiveCodecaveAddr = IntPtr.Zero;
+	vars.frameCodecaveAddr = IntPtr.Zero;
+
+	game.Resume(); // why not
```
Comment (ero-qt) [inline LOVE/LOVE_livesplit.asl]: Why?

File: LOVE/LOVE_livesplit.asl [hunk 3]
```diff
@@ -2,49 +2,214 @@ state("Love") {}
...
+
+	Thread.Sleep(1000); // hacky way to "fix" occasional update debug errors when you start the game. the errors seem harmless.
```
Comment (ero-qt) [inline LOVE/LOVE_livesplit.asl]: Sleeping UI thread. No-go.

File: LOVE/LOVE_livesplit.asl [hunk 4]
```diff
@@ -2,49 +2,214 @@ state("Love") {}
...
+				catch (Exception e)
+				{
+					game.Resume();
+					vars.Dbg(e.ToString());
+				}
+				finally
+				{
+					game.Resume();
+				}
+
```
Comment (ero-qt) [inline LOVE/LOVE_livesplit.asl:74]: Literally what is all of this? Since when do you ever need to write bytes?

File: LOVE/LOVE_livesplit.asl [hunk 5]
```diff
@@ -54,35 +219,103 @@ update
...
+		catch (Exception e)
+		{
+			game.Resume();
+			vars.Dbg(e.ToString());
+		}
+		finally
+		{
+			game.Resume();
+		}
+	}
```
Comment (ero-qt) [inline LOVE/LOVE_livesplit.asl]: Same here, what is this for? What reason could this *possibly* have?

File: LOVE/LOVE_livesplit.asl [hunk 6]
```diff
@@ -92,10 +325,61 @@ isLoading
 
 exit
 {
-	vars.CancelSource.Cancel();
+	if (vars.ScanThread.IsAlive) vars.CancelSource.Cancel();
```
Comment (ero-qt) [inline LOVE/LOVE_livesplit.asl]: Unnecessary check.

File: LOVE/LOVE_livesplit.asl [hunk 7]
```diff
@@ -92,10 +325,61 @@ isLoading
...
+	{
+		game.Resume();
+		vars.Dbg(e.ToString());
+	}
+	finally
+	{
+		game.Resume();
+	}
+
+	game.Resume(); // why not
```
Comment (ero-qt) [inline LOVE/LOVE_livesplit.asl]: Why?

**Comment by ero-qt** [comment]
Do not understand the changes. Lots of inconsistencies in casing for the variable names.

**Comment by ero-qt** [comment]
It's just commonly seen as really bad practice to inject code. I suppose if it works and is completely consistent, then that's fine, but how necessary is this? When was the game last updated? Will it ever be again? Is the game run on any version that isn't the current? Those are just the things you need to weigh to see if this is necessary.

If it is, then I'm fine with the changes and they can be merged.

**Comment by ero-qt** [comment]
Could merge at this point, no?

**Pull request merged**